constraints/miden-vm/memory.air

```diff
@@ -0,0 +1,78 @@
+mod MemoryAir
+
+### Helper functions/evaluators ###################################################################
+
+# Enforces that columns must be zero.
+ev equal_zero(&main: [4]):
+    enf v = 0 for v in &main
+
+# Enforces that values in the columns in the current row must be equal to the values in the next 
+# row.
+ev equal_seq_lines(&main: [4]):
+    enf v' = v for v in &main
+
+# Enforces that created flags and s[0] column have valid values during the program execution.
+ev flags_validity(main: [s_0], n0, n1, delta_a, delta_c):
+    # Enforce that n0 must be binary.
+    enf n0^2 = n0
+    # Enforce that when context changes, n0 = 1 (or when n0 = 0, context remains the same).
+    enf delta_c = 0 when !n0
+    # Enforce that n1 must be binary. An additional condition ensures that the check of n1 
+    # occurs only if the context does not change (n0 = 0).
+    enf n1^2 = n1 when !n0
+    # Enforce that if context remains the same, n1 = 1 when address changes and 0 otherwise.
+    enf delta_a = 0 when !n0 & !n1
+
+    # Enforce that values in the selectior columns must be binary.
+    # s[0] is set to 0 for write operations and to 1 for read operations.
+    enf col = col^2 for col in s
+
+# Enforces that selectors take the correct values under certain conditions.
+ev selectors_correctness(main: [s[2]], n0, n1):
+    # Enforce that s[1]' = 1 when the operation is a read and `ctx` and `addr` columns are both 
+    # unchanged.
+    enf s[1]' = 1 when !n0 & !n1 & s[0]'
+    # Enforce that when either the context changed, the address changed, or the operation is a 
+    # write, then s[1]' = 0
+    enf s[1]' = 0 when n0 | n1 | !s[0]'
+
+# Enforces that certain columns grow monotonically.
+ev monotonically_grow(n0, n1, delta_a, delta_c, delta_i):
```

Comment:
The signature looks incorrect. Should it be `ev monotonically_grow(main: [n0, n1, delta_a, delta_c, delta_i])`?

nit: Should we rename this to `grow_monotonically`?